arbeitszeit/repositories.py

```diff
@@ -532,3 +534,13 @@ def store_payout_factor(self, timestamp: datetime, payout_factor: Decimal) -> No
 
     def get_latest_payout_factor(self) -> Optional[PayoutFactor]:
         ...
+
+
+class DatabaseGateway(Protocol):
```

Comment:
The name `Repository` is not adequate in our case. Following clean architecture, a repository is supposed to get entities from some external data source. In our case the repositories are returning entities only in name. Using the terminology of uncle bob one should name our repositories "database gateway" instead. Our  "repositories" do merely return records but not entities.

In the wake of the new `*Result` concept the separation of the "repositories" is obsolete anyway since there will be only a handful of methods left when the refactoring is complete.